infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationServer.java

```diff
@@ -181,70 +235,41 @@ private void handleLogReplicationQueryLeadership(@Nonnull RequestMsg request,
+     * @param request Incoming request message
+     * @param ctx Channel context
+     * @param router Client router for sending the NACK
      */
-    protected synchronized boolean isLeader(@Nonnull RequestMsg request,
-                                            @Nonnull ChannelHandlerContext ctx,
-                                            @Nonnull IServerRouter router, boolean isLogEntry) {
-        // If the current cluster has switched to the active role (no longer the receiver) or it is no longer the leader,
-        // skip message processing (drop received message) and nack on leadership (loss of leadership)
-        // This will re-trigger leadership discovery on the sender.
-        boolean lostLeadership = isActive.get() || !isLeader.get();
-
-        if (lostLeadership) {
-
-            if (isLogEntry) {
-                LogReplicationEntryMsg entryMsg = request.getPayload().getLrEntry();
-                LogReplicationEntryType entryType = entryMsg.getMetadata().getEntryType();
-                log.warn("Received message of type {} while NOT LEADER. snapshotSyncSeqNumber={}, ts={}, syncRequestId={}", entryType,
-                        entryMsg.getMetadata().getSnapshotSyncSeqNum(), entryMsg.getMetadata().getTimestamp(),
-                        entryMsg.getMetadata().getSyncRequestId());
-            }
-
-            log.warn("This node has changed, active={}, leader={}. Dropping message type={}, id={}", isActive.get(),
-                    isLeader.get(), request.getPayload().getPayloadCase(), request.getHeader().getRequestId());
-            HeaderMsg responseHeader = getHeaderMsg(request.getHeader());
-            ResponseMsg response = getLeadershipLoss(responseHeader, localNodeId);
-            router.sendResponse(response, ctx);
-        }
-
-        return !lostLeadership;
+    private void sendLeadershipLoss(@Nonnull RequestMsg request,
+        @Nonnull ChannelHandlerContext ctx, @Nonnull IServerRouter router) {
+        HeaderMsg responseHeader = getHeaderMsg(request.getHeader());
+        ResponseMsg response = getLeadershipLoss(responseHeader, localNodeId);
+        router.sendResponse(response, ctx);
     }
 
-    /* ************ Public Methods ************ */
-
-    public synchronized void setLeadership(boolean leader) {
+    public void setLeadership(boolean leader) {
         isLeader.set(leader);
-    }
 
-    public void stopSink() {
-        sinkManager.stopOnLeadershipLoss();
-    }
-
-    public synchronized void setActive(boolean active) {
-        isActive.set(active);
+        // Reset the Sink Managers on acquiring leadership
+        if (isLeader.get()) {
+            sourceClientToSinkManagerMap.values().forEach(
+                sinkManager -> sinkManager.reset());
+        }
     }
 
-    public synchronized void setStandby(boolean standby) {
-        isStandby.set(standby);
+    public void stopSinks() {
```

Comment:
looks like this method is separately called after setting the leadership to false.  But yes, it can be done in setLeadership() itself, just like it is handled on acquiring leadership.